File: ruby/authentication-user-password.rb
require 'net/http'
require 'json'
require 'uri'

class ArtiaGerarToken
  ENDPOINT = 'https://api.artia.com/graphql'
  EMAIL = '<EMAIL_USER>'
  PASSWORD = '<PASSWORD>'

  def token
    uri = URI(ENDPOINT)
    http = Net::HTTP.new(uri.host, uri.port)
    http.use_ssl = true

    headers = { 'Content-Type' => 'application/json' }

    query = {
      query: <<~GRAPHQL,
        mutation {
          authenticationByEmail(
            email: "#{EMAIL}",
            password: "#{PASSWORD}"
          ) {
            token
          }
        }
      GRAPHQL
      variables: {}
    }

    request = Net::HTTP::Post.new(uri.path, headers)
    request.body = query.to_json

    response = http.request(request)

    if response.is_a?(Net::HTTPSuccess)
      body = JSON.parse(response.body)
      token = body.dig('data', 'authenticationByEmail', 'token')
      token
    else
      puts "Erro na requisição: #{response.code} - #{response.message}"
      nil
    end
  rescue => e
    puts "Erro inesperado: #{e.message}"
    nil
  end
end

gerador = ArtiaGerarToken.new
token = gerador.token
puts "Token final: #{token}" if token

File: javascript/authentication-user-password.js
const axios = require('axios');

const ENDPOINT = 'https://api.artia.com/graphql';
const EMAIL = '<EMAIL_USER>';
const PASSWORD = '<PASSWORD>';

async function gerarToken() {
  const headers = {
    'Content-Type': 'application/json',
  };

  const query = {
    query: `
      mutation {
        authenticationByEmail(
          email: "${EMAIL}",
          password: "${PASSWORD}"
        ) {
          token
        }
      }
    `,
    variables: {},
  };

  try {
    const response = await axios.post(ENDPOINT, query, { headers });

    if (response.status === 200) {
      const token = response.data?.data?.authenticationByEmail?.token;
      console.log("Token final:", token);
      return token;
    } else {
      console.error(`Erro na requisição: ${response.status} - ${response.statusText}`);
      return null;
    }
  } catch (error) {
    console.error("Erro inesperado:", error.message);
    return null;
  }
}

gerarToken();
